source/distributed/tria.cc

```diff
@@ -2808,6 +2832,32 @@ namespace parallel
                 && cell->level_subdomain_id() != this->locally_owned_subdomain())
               this->number_cache.level_ghost_owners.insert(cell->level_subdomain_id());
 
+#ifdef DEBUG
+          // Check that level_ghost_owners is symmetric by sending a message
+          // to every one
+          {
+
+            MPI_Barrier(this->mpi_communicator);
+
+            std::vector<MPI_Request> requests;
+            for (std::set<unsigned int>::iterator it = this->number_cache.level_ghost_owners.begin();
+                 it != this->number_cache.level_ghost_owners.end();
+                 ++it)
+              {
+                int dummy = 0;
+                requests.push_back(MPI_Request());
+                MPI_Isend(&dummy, 1, MPI_INT, *it, 9001, this->mpi_communicator, &requests.back());
+                requests.push_back(MPI_Request());
+                MPI_Irecv(&dummy, 1, MPI_INT, *it, 9001, this->mpi_communicator, &requests.back());
```

Comment:
Memory corruption foul! I was going to let you get by with using a local temporary variable in the `MPI_Isend` call (though you should really have been using a `MPI_IBsend` in that case). But doing so in `MPI_Irecv` is definitely a rookie mistake :-)